rest/server_context.go

```diff
@@ -1036,6 +1049,88 @@ func (sc *ServerContext) _removeDatabase(dbName string) bool {
 	return true
 }
 
+func (sc *ServerContext) isDatabaseSuspended(dbName string) bool {
+	sc.lock.RLock()
+	defer sc.lock.RUnlock()
+	return sc._isDatabaseSuspended(dbName)
+}
+
+func (sc *ServerContext) _isDatabaseSuspended(dbName string) bool {
+	if _, loaded := sc.databases_[dbName]; !loaded && sc.dbConfigs[dbName] != nil {
+		return true
+	}
+	return false
+}
+
+func (sc *ServerContext) suspendDatabase(dbName string) error {
+	sc.lock.Lock()
+	defer sc.lock.Unlock()
+
+	return sc._suspendDatabase(dbName)
+}
+
+func (sc *ServerContext) _suspendDatabase(dbName string) error {
+	dbCtx := sc.databases_[dbName]
+	if dbCtx == nil {
+		return base.ErrNotFound
+	}
+
+	if config, exists := sc.dbConfigs[dbName]; exists && !base.BoolDefault(config.Suspendable, sc.config.IsServerless()) {
+		return ErrSuspendingDisallowed
+	}
+
+	bucket := dbCtx.Bucket.GetName()
+	base.InfofCtx(context.TODO(), base.KeyAll, "Suspending db %q (bucket %q)", base.MD(dbName), base.MD(bucket))
+
+	if !sc._unloadDatabase(dbName) {
+		return base.ErrNotFound
+	}
+	delete(sc.bucketDbName, bucket)
+	return nil
+}
+
+func (sc *ServerContext) unsuspendDatabase(dbName string) (*db.DatabaseContext, error) {
+	sc.lock.Lock()
+	defer sc.lock.Unlock()
+
+	return sc._unsuspendDatabase(dbName)
+}
+func (sc *ServerContext) _unsuspendDatabase(dbName string) (*db.DatabaseContext, error) {
```

Comment:
It looks like the only callers of _unsuspendDatabase and _suspendDatabase are the locking versions.  If the non-locking versions aren't needed otherwise, we can remove.

Reply:
These are used in CBG-2306